src/Solver/time_stepping/TimeManager.cpp

```diff
@@ -378,3 +382,7 @@ void seissol::time_stepping::TimeManager::setTv(double tv) {
   }
 }
 
+void seissol::time_stepping::TimeManager::resetCommunicationManager() {
+  communicationManager.reset(nullptr);
+}
```

Comment:
reset accepts a double and not a pointer. Also not sure why this function is neccessary.

Reply:
Hi @uphoffc , I didn't get why it should accept `double`. `communicationManager` is a unique pointer. Calling reset will trigger the destructor of the underlying object.

Note, the `SeisSol` object is static and thus it is going to be deleted only at the very very end. At this point, the connection with the GPU driver will be lost. Omitting a call to the dtor of `communicationManager` will trigger a segfault because `WithCopy` implementations uses device memory which is freed by the dtor

Reply:
Ok sorry about that. I saw that AbstractCommunicationManager has a reset function and did not notice that it is stored as unique_ptr in TimeManager.